processor/src/main/java/org/mapstruct/ap/internal/model/source/SourceMethod.java

```diff
@@ -238,7 +241,11 @@ private SourceMethod(Type declaringMapper, ExecutableElement executable, List<Pa
         this.mapperToImplement = mapperToImplement;
     }
 
-    private Parameter determineMappingTargetParameter(Iterable<Parameter> parameters) {
+    private boolean determineIfIsObjectFactory(ExecutableElement executable) {
+		return executable.getAnnotation(ObjectFactory.class) != null || getTargetTypeParameter() != null || parameters.isEmpty();
```

Comment:
Use the `ObjectFactoryPrism` here to determine if the executable is annotated with `ObjectFactory`.

Rule of thumb: we don't use any classes from the annotation modules (`mapstruct` and `mapstruct-jdk8`) in the processor (with the exception of the reference in the generator class for the prisms). That way the processor can run without having the annotations on its own classpath. And it usually doesn't have them, which would explain your `NoClassDefFound` errors.